### PR TITLE
ADAPT-4216: Port Vault variable management strategy.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Ports vault variable management strategy from give.stanford.edu

# Review By (Date)
- Whenever

# Criticality
- Low

# Review Tasks

## Setup tasks and/or behavior to test

1. Review code
2. Review preview build website for irregularities
3. Check out branch and follow installation instructions to get a local site up and running
4. Review vault.stanford.edu and the giving secrets for any irregularities
5. Review readme instructions
6. Run installation steps and validate you get a running local environment

### CODE REVIEW

**Review by mattanglin — APPROVED** [review]
lgtm! 👍🏼

**Comment by sherakama** [comment]
@nplowman please have a review as well as this will cross over on your work with the auth-sdk.

File: README.md [hunk 1]
```diff
@@ -12,67 +12,109 @@ Description
...
+_Development_
 
-*Development*
 ```
-cp example.env .env.development
-* Get the Storyblok preview access token *
+cp example.env .env
+* Manually add the vault secret and role id to .env
+* You can get the variables by running `netlify login` then `netlify link` and then `netlify env`
+* After the VAULT_ROLE_ID and VAULT_SECRET_ID environment variables have been set run
```
Comment (yvonnetangsu) [inline README.md:23]:
```suggestion
* You can also get the vault secret and role id by first running `netlify login` then `netlify link`.
* Then use the `netlify env:get VAR_NAME` and `netlify env:set VAR_NAME value` commands to retrieve and set the `VAULT_ROLE_ID` and `VAULT_ROLE_ID`.
* After the `VAULT_ROLE_ID` and `VAULT_SECRET_ID` environment variables have been set,
```
Super minor suggestions to make this a bit more straightforward

File: README.md [hunk 2]
```diff
@@ -12,67 +12,109 @@ Description
...
 * Get the FontAwesome NPM token from the Netlify site settings in environment variables *
```
Comment (yvonnetangsu) [inline README.md:24]:
```suggestion
get the FontAwesome NPM token from the Netlify site settings in environment variables *
```
Just trying to delete this bullet so it reads as one sentence with the line above:
"After the `VAULT_ROLE_ID` and `VAULT_SECRET_ID` environment variables have been set, get the FontAwesome NPM token from the Netlify site settings in environment variables"

**Review by yvonnetangsu — APPROVED** [review]
I pulled down a fresh repo and followed all the steps to set up a working local. Also checked Alumni and T/S site - everything looks great! Pre-approving with super minor suggested edits for the readme - totally optional 😄  Thanks, @sherakama !

**Review by nplowman — APPROVED** [review]
@sherakama This looks good to me. 
Didn't realize there was another ticket for this already - I had actually copy-pasted a lot of this code into my feature branch already as I needed the Vault secrets for development and testing.  (This is a bit more complete than what I have though, so makes sense to merge this, and I can rebase).

**Pull request merged**